test/e2e/scrapeconfig_test.go

```diff
@@ -491,3 +491,25 @@ func testScrapeConfigDNSSDConfig(t *testing.T) {
 	err = framework.WaitForActiveTargets(context.Background(), ns, "prometheus-operated", 0)
 	require.NoError(t, err)
 }
+
+// testScrapeConfigDNSSDConfigWithWrongRecordType tests whether DNS SD based monitoring rejects invalid DNS record type.
+func testScrapeConfigDNSSDConfigWithWrongRecordType(t *testing.T) {
```

Comment:
Of course! I have done the same with K8S. Once that is merged, I think it would be easier for me rather than having to deal with potential merge conflicts.